templates/dotnet/src/Appwrite/Services/ServiceTemplate.cs.twig

```diff
@@ -1,6 +1,6 @@
 {% macro parameter(parameter) %}
-{% if parameter.name == 'orderType' %}{{ 'OrderType orderType' }}{% else %}
-{{ parameter.type | typeName }} {{ parameter.name | caseCamel | escapeKeyword }}{{ parameter | escapeDollarSign | paramDefault }}{% endif %}
+{% if parameter.name == 'orderType' %}{{ 'OrderType orderType = OrderType.ASC' }}{% else %}
+{{ parameter.type | typeName }}{% if not parameter.required and (parameter.type == 'boolean' or parameter.type == 'integer') %}?{% endif %} {{ parameter.name | caseCamel | escapeKeyword }}{{ parameter | escapeDollarSign | paramDefault }}{% endif %}
```

Comment:
Why do we have a special condition for booleans and ints?

Reply:
They only allow `null` values with a `?`.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types